docs/framework/react/guides/custom-errors.md

```diff
@@ -0,0 +1,284 @@
+---
+id: custom-errors
+title: Custom Errors
+---
+
+TanStack Form provides complete flexibility in the types of error values you can return from validators. While string errors are common and easy to work with, the library supports any type of value as an error.
+
+## Return String Values from Forms
+
+```tsx
+<form.Field
+  name="username"
+  validators={{
+    onChange: ({ value }) =>
+      value.length < 3 ? 'Username must be at least 3 characters' : undefined,
+  }}
+/>
+```
+
+For form-level validation affecting multiple fields:
+
+```tsx
+const form = useForm({
+  defaultValues: {
+    username: '',
+    email: '',
+  },
+  validators: {
+    onChange: ({ value }) => {
+      return {
+        fields: {
+          username:
+            value.username.length < 3 ? 'Username too short' : undefined,
+          email: !value.email.includes('@') ? 'Invalid email' : undefined,
+        },
+      }
+    },
+  },
+})
+```
+
+String errors are the most common type and are easily displayed in your UI:
+
+```tsx
+{
+  field.state.meta.errors.map((error, i) => (
+    <div key={i} className="error">
+      {error}
+    </div>
+  ))
+}
+```
+
+## Return Non-String Values from Forms
+
+TanStack Form supports returning non-string values as errors, providing flexibility for different validation scenarios. It's important to note that any `truthy` value is handled as an error, while falsy values (except for 0) are considered valid. For example, returning 0 is treated as valid, not as an error.
```

Comment:
I'm not sure about this paragraph. 0 is in fact falsy, it's not an exception.

Since this information now is also at the beginning of the page, what you think about removing this line entirely here and adding the truthy=error falsy=no error clarification at the top?

Reply:
agree, will remove that and put it at the top like this 
as a note truthy values will return the error while falsy will not
`truthy = error`
`falsy = no error`